plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/ActiveMapTracker.java

```diff
@@ -79,15 +79,26 @@
      * @return
      */
     public Map getActiveMap() {
+        MapPart activePart = getActiveMapPartInternal();
+        if (activePart != null) {
+            return activePart.getMap();
+        }
 
-        if (Display.getCurrent() != null) {
-            MapPart part = getActivePart();
-            if (part != null && !activeParts.isEmpty() && part != activeParts.get(0)) {
-                addActivePart(part);
-                return part.getMap();
-            }
+        MapPart visibleMapPart = getFirstVisibleMapPart();
+        return (visibleMapPart == null ? ApplicationGIS.NO_MAP : visibleMapPart.getMap());
+    }
+
+    /**
+     * @return most recent opened MapPart or null.
+     */
+    public MapPart getMostRecentOpenedPart() {
+        if (!openMaps.isEmpty()) {
+            return openMaps.get(0);
         }
+        return null;
```

Comment:
thats the reason why it isn't a set anymore. Wondering if a Stack would fit much better here